crate.go

```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"time"
+
+	"github.com/go-kit/kit/endpoint"
+	"github.com/jackc/pgx"
+	"github.com/jackc/pgx/pgtype"
+	"github.com/prometheus/common/model"
+)
+
+type crateRow struct {
+	labels     model.Metric
+	labelsHash string
+	timestamp  time.Time
+	value      float64
+	valueRaw   int64
+}
+
+type crateWriteRequest struct {
+	rows []*crateRow
+}
+
+type crateReadRequest struct {
+	stmt string
+}
+
+type crateReadResponse struct {
+	rows []*crateRow
+}
+
+type crateEndpoint struct {
+	pool     *pgx.ConnPool
+	poolConf pgx.ConnPoolConfig
+}
+
+func newCrateEndpoint(ep *endpointConfig) *crateEndpoint {
+	connConf := pgx.ConnConfig{
+		Host:     ep.Host,
+		Port:     ep.Port,
+		User:     ep.User,
+		Password: ep.Password,
+		Database: ep.Database,
```

Comment:
`Database: ep.Schema,`

CrateDB does not know the concept of a database in terms of other RDBMSs.
We should always say `schema`, if the equivalent of database in Postgres is meant.

Reply:
Good to know, will call it schema on our end (can't change the `pgx.ConnConfig` struct of course).